tests/Backend/Mixed/SharingTest.php

```diff
@@ -1170,6 +1170,363 @@ function (array $column) {
+            $bucketId,
+            [
+                'include' => 'columnMetadata'
+            ]
+        );
+
+        $aliasAliasTableId = $bucketId . '.tableAliasAlias';
+        $tables = array_values(array_filter($response['tables'], function ($table) use ($aliasAliasTableId) {
+            return $table['id'] === $aliasAliasTableId;
+        }));
+
+        $this->assertNotEmpty($tables[0]['columnMetadata']['id']);
+        $this->assertEquals(
+            $mdList,
+            $tables[0]['columnMetadata']['id']
+        );
+    }
+
+    /**
+     * @dataProvider sharingBackendData
+     * @throws ClientException
+     */
+    public function testGetSharedBucketDetailForNonExistBucket($backend)
+    {
+        $this->initTestBuckets($backend);
+        $bucketId = reset($this->_bucketIds);
+
+        $tableName = 'transactions';
+        $this->_client->createTable(
+            $bucketId,
+            $tableName,
+            new CsvFile(__DIR__ . '/../../_data/transactions.csv')
+        );
+
+        $this->_client->shareBucket($bucketId);
+        $this->assertTrue($this->_client->isSharedBucket($bucketId));
+
+        $response = $this->_client->verifyToken();
+        $this->assertArrayHasKey('owner', $response);
+
+        $this->assertArrayHasKey('id', $response['owner']);
+        $this->assertArrayHasKey('name', $response['owner']);
+
+        $project = $response['owner'];
+
+        try {
+            $this->_client2->getSharedBucketDetail(
+                $project['id'],
+                'non-exist',
+                [
+                    'include' => 'columns'
+                ]
+            );
+        } catch (ClientException $e) {
+            $this->assertEquals('storage.buckets.notFound', $e->getStringCode());
+            $this->assertEquals(404, $e->getCode());
+        }
+    }
+
+    /**
+     * @dataProvider sharingBackendData
+     * @throws ClientException
+     */
+    public function testShareBucketDetailVisibility($backend)
+    {
+        $this->initTestBuckets($backend);
+
+        $newBucketId = self::STAGE_IN . '.c-visibility-test';
+        foreach ($this->_client2->listBuckets() as $bucket) {
+            if ($bucket['id'] === $newBucketId) {
+                $this->_client2->dropBucket($newBucketId);
+            }
+        }
+        $client2BucketId = $this->_client2->createBucket('visibility-test', self::STAGE_IN, 'Api tests', $backend);
+        $clientEmail = $this->_client->verifyToken()['description'];
+        $baseProjectId = $this->_client->verifyToken()['owner']['id'];
+        $client2ProjectId = $this->_client2->verifyToken()['owner']['id'];
+
+        $this->_client2->shareBucketToProjects($client2BucketId, $baseProjectId);
+
+        $response = $this->_client->getSharedBucketDetail($client2ProjectId, $newBucketId);
+        $this->assertNotEmpty($response);
+
+        try {
+            $this->_client2->getSharedBucketDetail($client2ProjectId, $newBucketId);
```

Comment:
Co testuje tohle? Čekal bych,ž e to bude testovat, že jiný projekt to nevidí. Ale v tomhle případě to nevidí protože je to jeho bucket v jeho projektu, ne?